plugins/CorePluginsAdmin/Controller.php

```diff
@@ -445,7 +377,22 @@ public function activate($redirectAfter = true)
             $notification->context = Notification::CONTEXT_SUCCESS;
             Notification\Manager::notify('CorePluginsAdmin_PluginActivated', $notification);
 
-            $this->redirectToIndex('CorePluginsAdmin', $actionToRedirect);
+            $redirectTo = Common::getRequestVar('redirectTo', '', 'string');
+            if (!empty($redirectTo) && $redirectTo === 'marketplace') {
+                $this->redirectToIndex('Marketplace', 'overview');
```

Comment:
Eg when activating a plugin in the Marketplace we want to make sure to be still on the Marketplace page afterwards as someone might want to install or activate another plugin